e2e/test/scenarios/admin/performance/dashboardsAndQuestions.cy.spec.ts

```diff
@@ -0,0 +1,320 @@
+            strategy: sampleAdaptiveStrategy,
+            item: sampleDashboard,
+          },
+          {
+            description:
+              "dashboards can inherit adaptive strategy from database",
+            strategy: sampleAdaptiveStrategy,
+            item: sampleDashboard,
+            inheritsStrategyFrom: sampleDatabase,
+          },
+          {
+            description:
+              "dashboards can inherit adaptive strategy from instance-wide default policy",
+            strategy: sampleAdaptiveStrategy,
+            item: sampleDashboard,
+            inheritsStrategyFrom: instanceDefault,
+          },
+          {
+            description: "dashboards can use duration strategy",
+            strategy: sampleDurationStrategy,
+            item: sampleDashboard,
+          },
+          {
+            description:
+              "dashboards can inherit duration strategy from database",
+            strategy: sampleDurationStrategy,
+            item: sampleDashboard,
+            inheritsStrategyFrom: sampleDatabase,
+          },
+          {
+            description:
+              "dashboards can inherit duration strategy from instance-wide default policy",
+            strategy: sampleDurationStrategy,
+            item: sampleDashboard,
+            inheritsStrategyFrom: instanceDefault,
+          },
+          // FIXME: Add other kinds of tests:
+          // - When question runtime is less than minimum query duration, adaptive strategy does not apply
+        ] as CacheTestParameters[]
+      ).forEach(
+        ({
+          strategy,
+          item,
+          inheritsStrategyFrom,
+          description,
+          it = itFunction,
+        }) => {
+          it(description, () => {
+            const { reload, visitItem } =
+              item.model === "question"
+                ? setupQuestionTest(sampleQuestion)
+                : setupDashboardTest(sampleDashboard, sampleQuestion);
+
+            const itemToConfigure = inheritsStrategyFrom ?? item;
+
+            cy.then(function () {
+              selectCacheStrategy({
+                item: itemToConfigure,
+                strategy,
+              });
+
+              visitItem();
+            });
+
+            cy.then(function () {
+              reloadUntil(resultIsCached, {
+                logMessage: "Reloading until the result is cached",
+                wrapFinalResultAs: "cachedResult",
+                wrapTimeOfFinalResultAs: "firstCachedResultSeenAt",
+                reload,
+              });
+            });
+
+            const expectedCacheDuration =
+              getMillisecondsUntilCacheShouldBeInvalidated(strategy);
+            cy.then(function () {
+              getCacheDuration({
+                cacheStartTime: this.firstCachedResultSeenAt,
+                expectedCacheDuration,
+                reload,
+                questionRuntime,
+              }).as("cacheDuration");
+            });
```

Comment:
One thing I learned while writing these tests is that previously wrapped values (i.e. values wrapped under aliases) are available in the next `cy.then` block as `this.aliasName`. I use this pattern frequently to avoid nested callbacks.